src/components/panel/panel.spec.js

```diff
@@ -1475,6 +1696,29 @@ describe('$mdPanel', function() {
   }
 
   /**
+   * Attaches the panel. If a config value is passed, creates a new panelRef
+   * using $mdPanel.open(config); Otherwise, called open on the panelRef,
+   * assuming one has already been created.
+   * @param {!Object=} opt_config
+   */
+  function attachPanel(opt_config) {
+    var attachPromise; 
+    if (opt_config) {
+      panelRef = $mdPanel.create(opt_config);
+    }
+
+    if (panelRef) {
+      attachPromise = panelRef.attach();
+    }
+
+    $rootScope.$apply();
+    $rootScope.$apply();
+    $material.flushOutstandingAnimations();
+
+    return attachPromise;
```

Comment:
You shouldn't return the promise here. If you need a promise in the tests, you can use the real call in that test. (The `$rootScope.$apply()` resolves the attachPromise, so even if it's returned, you can't _do_ anything with it.)